src/check_wheel_contents/checker.py

```diff
@@ -299,6 +299,29 @@ def check_W010(self, contents: WheelContents) -> list[FailedCheck]:
         else:
             return []
 
+    def check_W011(self, contents: WheelContents) -> list[FailedCheck]:
+        """
+        W011 — Wheel contains module and submodule with the same name
+
+        E.g: checks for both `path/foo.py` and `path/foo/__init__.py` being present
+        """
+        conflicts: list[str] = []
+
+        def check_dir(tree: Directory) -> None:
+            for name, entry in tree.entries.items():
+                if isinstance(entry, Directory):
+                    if tree.files.get(name + ".py") is not None:
+                        conflicts.append(entry.path or name)
+                    check_dir(entry)
+
+        for tree in (contents.purelib_tree, contents.platlib_tree):
+            if isinstance(tree, Directory):
```

Comment:
This line is redundant; `contents.purelib_tree` and `contents.platlib_tree` are always `Directory`s.